docs/api-spec.md

```diff
@@ -48,269 +72,454 @@ There is no formal specification of the Kubernetes API and Resource Model. This
+| `image`      | string                | REQUIRED    |       |
+| `args`       | []string              | REQUIRED    |       |
+| `command`    | []string              | REQUIRED    |       |
+| `workingDir` | []string              | REQUIRED    |       |
+| `env`        | [][`EnvVar`](#envvar) | REQUIRED    |       |
+| `script`     | string                | REQUIRED    |       |
 
-| Field                 | Type   | Requirement |
-|-----------------------|--------|-------------|
-| `type`                | string | REQUIRED    |
-| `status`              | Enum:<br>- `"True"`<br>- `"False"`<br>- `"Unknown"` (default) | REQUIRED |
-| `reason`              | string | REQUIRED    |
-| `message`             | string | REQUIRED    |
-| `severity`            | Enum:<br>- `""` (default)<br>- `"Warning"`<br>- `"Info"` | REQUIRED |
-| `lastTransitionTime`* | string | OPTIONAL    |
+### Sidecar
 
-\* If `lastTransitionTime` is populated by the implementation, it must be in [RFC3339](https://tools.ietf.org/html/rfc3339).
+Specifies a list of containers to run alongside the Steps in a Task. If sidecars are supported, the following fields are required:
 
-Additionally, the resource's `status.conditions` field MUST be managed as follows to enable clients to present useful diagnostic and error information to the user.
+| Field             | Type                                  | Requirement | Notes                                                                                                                       |
+|-------------------|---------------------------------------|-------------|-----------------------------------------------------------------------------------------------------------------------------|
+| `name`            | string                                | REQUIRED    | Name of the Sidecar specified as a DNS_LABEL. Each Sidecar in a Task must have a unique name (DNS_LABEL).Cannot be updated. |
+| `image`           | string                                | REQUIRED    | [Container image name](https://kubernetes.io/docs/concepts/containers/images/#image-names)                                  |
+| `command`         | []string                              | REQUIRED    | Entrypoint array. Not executed within a shell. The image's ENTRYPOINT is used if this is not provided.                      |
+| `args`            | []string                              | REQUIRED    | Arguments to the entrypoint. The image's CMD is used if this is not provided.                                               |
+| `script`          | string                                | REQUIRED    | Script is the contents of an executable file to execute. If Script is not empty, the Sidecar cannot have a Command or Args. |
+| `securityContext` | [`SecurityContext`](#securitycontext) | REQUIRED    | Defines the security options the Sidecar should be run with.                                                                |
```

Comment:
Why does `Sidecar` have a REQUIRED `securityContext` and `Step` has no `securityContext`  instead?

Reply:
I think this was initially done for supporting the docker in docker sidecar use case where the sidecar daemon needs privilege but not the step. But I think that might be too narrow. I think, I'm going to add security context to both step and sidecars.